### PR TITLE
Cli next preview (NO MERGE)

### PR DESCRIPTION
**Pull request by heyMP**
relates #59 

- Migrated from Commander.js to Open CLI Framework
- Migrated init generator to cli.
- Playing around with Listr

### CODE REVIEW

**Comment by heyMP** [comment]
kapture 2018-10-02 at 9 58 31

File: packages/cli/package.json [hunk 1]
```diff
@@ -1,39 +1,75 @@
...
-  "author": "",
   "license": "Apache-2.0",
-  "dependencies": {
-    "@wcfactory/generator-wcfactory": "^0.1.8",
-    "commander": "^2.18.0",
-    "please-upgrade-node": "^3.1.1",
-    "update-notifier": "^2.5.0",
-    "yeoman-environment": "^2.3.3"
+  "main": "lib/index.js",
+  "oclif": {
```
Comment (ChristianMurphy) [inline packages/cli/package.json]: Interesting, this is the first time I've heard of Open CLI.
Most CLI project's I've seen use http://yargs.js.org, and more recently there has been an uptick in the use of https://www.npmjs.com/package/ink

How are you liking Open CLI so far?

Comment (heyMP) [reply packages/cli/package.json]: oooo I have not seen ink.  That definitely looks interesting.  Open CLI is good but it's not doing a whole lot.  It's one level above yargs but in my opinion I would prefer something like ink.  Ink looks like we can customize the output to a much greater extent.

In this pull request I have Open CLI, Inquirer.js, Listr smooshed together.  I'm finding that Inquirer and Listr get me 80% there but it's going to be tough to get the next 20%.

What I will probably do is finish this pull request with Open CLI and remove any of the stuff that isn't fully working.  Then we can take a look at Ink or something similar to give us full control over the output.

File: packages/common/wcflibraries.js [hunk 2]
```diff
@@ -39,5 +38,17 @@ const wcfLibrariesListChoices = () => {
...
+/**
+ * Current state of the wcf cache file.
+ * @return object
+ */
+const wcfLibrariesCache = (() => {
+  const cacheFile = `${cwd}/.wcflibcache.json`;
+  if (fs.existsSync(cacheFile)) {
+    return JSON.parse(fs.readFileSync(`${cwd}/.wcflibcache.json`, 'utf8'))
+  }
+  return {}
```
[the remark this answers is not in the document]
Comment (heyMP) [reply packages/common/wcflibraries.js]: go ahead and modify that function.  I broke this stuff out into smaller functions to make them easier to modify.  The only time these have been breaking the cli is trying to require or read a file that doesn't exist.

**Pull request merged**